Python/fhirabstractbase.py

```diff
@@ -102,11 +102,25 @@ def update_with_json(self, jsondict, cast=False):
             if is_list and not isinstance(jsondict[jsname], list):
                 raise TypeError("The field '%s' must be a list" % jsname)
 
+            if not self._validate_element_value(name, jsondict[jsname]):
```

Comment:
I'm not sure why per-element validation is called before global type checks and cast? Could there be type mismatch? Should `_validate_element_value` check for lists itself? And so on.